go-controller/pkg/ovn/ovn.go

```diff
@@ -660,6 +668,12 @@ func (oc *Controller) ensurePod(oldPod, pod *kapi.Pod, addPort bool) bool {
 		oc.deletePodExternalGW(oldPod)
 	}
 
+	if util.PodCompleted(pod) {
```

Comment:
also note ensurePods is getting a bit overloaded? we claim to setup a pod here but are deleting pod setups,
so q, for completed pods, do we really need to do this in the addHandler? Meaning a new pod can't really be in completed state right? Isn't it enough we cover it in the updateHandler? -> my understanding is this is what we do for egressIPs code path as well.

Reply:
to your first comment, yeah we should avoid the double call to deletePodExternalGW.

2nd comment I see your point, but what happens if ovnkube-master is restarted. Wont all the of the existing pods get an "add" and they could have been moved to completed state while ovnkube-master was down?

Reply:
> to your first comment, yeah we should avoid the double call to deletePodExternalGW.
> 
> 2nd comment I see your point, but what happens if ovnkube-master is restarted. Wont all the of the existing pods get an "add" and they could have been moved to completed state while ovnkube-master was down?

yeah! good point. Also now that I was looking at my PR https://github.com/ovn-org/ovn-kubernetes/pull/2787 I realized its easier to keep it in the `ensurePods` to include the retry logic, but now I am a bit confused: what will happen if deleteLogicalPort for completed pod fails here? Will it be a add retry or delete retry? I hope its an add retry right? else it will screw too much with  my logic.

Reply:
I just changed all this and removed this logic from ensurePod. Moved it to the actual handler itself to be more inline with your delete retry logic.